pkg/operator/rules.go

```diff
@@ -42,31 +43,50 @@ const (
 
 type PrometheusRuleSelector struct {
 	ruleFormat   RuleConfigurationFormat
+	version      semver.Version
 	ruleSelector labels.Selector
 	nsLabeler    *namespacelabeler.Labeler
 	ruleInformer *informers.ForResource
 	logger       log.Logger
 }
 
-func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, labelSelector *metav1.LabelSelector, nsLabeler *namespacelabeler.Labeler, ruleInformer *informers.ForResource, logger log.Logger) (*PrometheusRuleSelector, error) {
+func NewPrometheusRuleSelector(ruleFormat RuleConfigurationFormat, version string, labelSelector *metav1.LabelSelector, nsLabeler *namespacelabeler.Labeler, ruleInformer *informers.ForResource, logger log.Logger) (*PrometheusRuleSelector, error) {
+	componentVersion, err := semver.ParseTolerant(version)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse Prometheus version")
+	}
+
 	ruleSelector, err := metav1.LabelSelectorAsSelector(labelSelector)
 	if err != nil {
 		return nil, errors.Wrap(err, "convert rule label selector to selector")
 	}
 
 	return &PrometheusRuleSelector{
 		ruleFormat:   ruleFormat,
+		version:      componentVersion,
 		ruleSelector: ruleSelector,
 		nsLabeler:    nsLabeler,
 		ruleInformer: ruleInformer,
 		logger:       logger,
 	}, nil
 }
 
-func generateRulesConfiguration(ruleformat RuleConfigurationFormat, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
-	if ruleformat == PrometheusFormat {
-		// Unset partialResponseStrategy field.
-		for i := range promRule.Groups {
+func generateRulesConfiguration(ruleformat RuleConfigurationFormat, version semver.Version, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
+	minVersionLimits := semver.MustParse("2.31.0")
+	component := "Prometheus"
+	if ruleformat == ThanosFormat {
+		minVersionLimits = semver.MustParse("0.24.0")
+		component = "Thanos"
+	}
+
+	for i := range promRule.Groups {
+		if promRule.Groups[i].Limit != 0 {
+			if version.LT(minVersionLimits) {
+				level.Warn(logger).Log("msg", fmt.Sprintf("`limit` is supported only from %s version %q", component, minVersionLimits))
+			}
+		}
+		if component == "Prometheus" {
```

Comment:
better to use the const value
```suggestion
		if cr.ruleFormat == PrometheusFormat {
```